packages/shared-ux/chrome/navigation/src/ui/v2/components/navigation.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, {
+  createContext,
+  useState,
+  useCallback,
+  ReactNode,
+  useMemo,
+  useEffect,
+  useContext,
+  useRef,
+} from 'react';
+
+import { useNavigation as useNavigationServices } from '../../../services';
+import { InternalNavigationNode, RegisterFunction } from '../types';
+import { NavigationGroup } from './navigation_group';
+import { NavigationItem } from './navigation_item';
+
+interface Context {
+  register: RegisterFunction;
+}
+
+const NavigationContext = createContext<Context>({
```

Comment:
nit: I think `Context` is too generic and I would like to suggest giving the interface the same name as the value:
```suggestion
interface NavigationContext {
  register: RegisterFunction;
}

const NavigationContext = createContext<NavigationContext>({
```

Reply:
I usually prefer to keep `Context` when it is not exported as it is clear it is the context of the current file. Having 2 different "things" (a React Context and a Typescript interface) using the same name would be more confusing to me. 😊